File: easy-projects/easy-common/src/main/java/xyz/srclab/common/exception/DefaultExceptionStatus.java
package xyz.srclab.common.exception;

import xyz.srclab.annotation.Nullable;

public enum DefaultExceptionStatus implements ExceptionStatus {

    INTERNAL("000000", "Internal Error"),
    ;

    private final ExceptionStatus impl;

    DefaultExceptionStatus(String code, @Nullable String description) {
        this.impl = new ExceptionStatusImpl(code, description);
    }

    @Override
    public String getCode() {
        return impl.getCode();
    }

    @Override
    @Nullable
    public String getDescription() {
        return impl.getDescription();
    }

    @Override
    public String toString() {
        return impl.toString();
    }
}

File: srclab-common/src/main/kotlin/xyz/srclab/common/exception/DefaultExceptionStatus.kt
package xyz.srclab.common.exception

import xyz.srclab.common.state.stateToString

enum class DefaultExceptionStatus(
    @get:JvmName("code") override val code: String,
    @get:JvmName("description") override val description: String?
) : ExceptionStatus {

    INTERNAL("000000", "Internal Error"),
    UNKNOWN("000001", "Unknown Error"),
    ;

    override fun toString(): String {
        return stateToString()
    }
}